### ISSUE TITLE
touchSpin supports only Double

### ISSUE THREAD

**Issue by NicolaIsotta** [TheCoder4eu/BootsFaces-OSP#550]
I was trying the new touchSpin component. When using an Integer as the value, an excepton is raised by these code lines in net.bootsfaces.component.touchSpin.touchSpin (line 82):

```
	@Override

	public String getValue() {

		Object val = getStateHelper().eval(PropertyKeys.value);

		if(val instanceof Double) return val.toString();

		else return (String) val;

	}
```

In my case, replacing `instanceof Double` with `instanceof Number` solved the issue.

**Comment by zhedar** [comment]
That's why I'm an advocate for using `Number` instead of `Integer` or `Double` for such components (see https://github.com/stephanrauh/JSFLibraryGenerator/issues/3). Any new thoughts on that now @stephanrauh?

**Comment by stephanrauh** [comment]
OK, let's do it. We need to fix the generator, anyways. For some reason it's always locked in an infinite loop.

**Comment by asterd** [comment]
Hi @NicolaIsotta, in my last commit (a71270f) I've fixed this error with your suggestion. 
@stephanrauh @zhedar I've not changed the generator to support Number because the Number class is an high-level abstraction of the primitive wrappers that need a typified conversion. So, you must always define the real type of the item during conversion. So, I've decided to change the type cast of this single method, as suggested by Nicolalsotta instead of introducing a big change that needs a lot of tests and (to me) is not a real useful change.
